[package] libmp3lame/3.100: Missing support for Windows on ARM64

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libmp3lame/3.100**
  * Operating System+version: **Windows 11 21H2**
  * Compiler+version: **Visual Studio 2022**
  * Conan version: **conan 1.48.0**
  * Python version: **Python 3.9.12**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=armv8
```

### Steps to reproduce (Include if Applicable)

Build Audacity (current master) from the VS x64_ARM64 developer prompt ([vcvarsall.bat] Environment initialized for: 'x64').

Eventually, after successfully compiling a lot of of other libs that Audacity depends on it dies like this:
```
--- HIP DECODING LIBRARY UPTODATE ---

libmp3lame\bitstream.obj : fatal error LNK1112: module machine type 'ARM64' conflicts with target machine type 'x86'
libmp3lame/3.100: 
```

--- a/recipes/libmp3lame/all/conanfile.py
+++ b/recipes/libmp3lame/all/conanfile.py
@@ -91,7 +91,7 @@ class LibMP3LameConan(ConanFile):
             tools.replace_in_file("Makefile.MSVC", " /GL", "")
             tools.replace_in_file("Makefile.MSVC", " /LTCG", "")
             tools.replace_in_file("Makefile.MSVC", "ADDL_OBJ = bufferoverflowU.lib", "")
-            command = "nmake -f Makefile.MSVC comp=msvc asm=yes"
+            command = "nmake -f Makefile.MSVC comp=msvc"
             if self._is_clang_cl:
                 cl = os.environ.get('CC', "clang-cl")
                 link = os.environ.get("LD", 'lld-link')
@@ -103,7 +103,12 @@ class LibMP3LameConan(ConanFile):
                 tools.replace_in_file('Makefile.MSVC', '/GAy', '/GA')
             if self.settings.arch == "x86_64":
                 tools.replace_in_file("Makefile.MSVC", "MACHINE = /machine:I386", "MACHINE =/machine:X64")
+                command += " MSVCVER=Win64 asm=yes"
+            elif self.settings.arch == "armv8":
+                tools.replace_in_file("Makefile.MSVC", "MACHINE = /machine:I386", "MACHINE =/machine:ARM64")
                 command += " MSVCVER=Win64"
+            else:
+                command += " asm=yes"
             command += " libmp3lame.dll" if self.options.shared else " libmp3lame-static.lib"
             self.run(command)
 

